### PR TITLE
GFB-8 Allow to give content for the unsaved files

### PR DESCRIPTION
**Pull request by serhat-yenican-sonarsource**

### CODE REVIEW

File: src/main/java/org/sonar/scm/git/blame/BlobReader.java [hunk 1]
```diff
@@ -81,14 +89,24 @@ private RawText loadText(String path) throws IOException {
     if (treeWalk.next()) {
       FileTreeIterator iter = treeWalk.getTree(0, FileTreeIterator.class);
       if ((iter.getEntryRawMode() & TYPE_MASK) == TYPE_FILE) {
-        try (InputStream is = iter.openEntryStream()) {
-          return new RawText(is.readAllBytes());
+        String fileContent = fileContents.get(path);
+        if(fileContent != null) {
```
Comment (henryju) [inline src/main/java/org/sonar/scm/git/blame/BlobReader.java:93]:
```suggestion
        if (fileContent != null) {
```

**Review by henryju — COMMENTED** [review]
I find it dangerous to pass a `Map<String, String>`. This give no other option to the caller to pre-load all content in memory.
Why not using a `Function<String, String>` instead, so that caller can lazily load the content if possible?

**Review by henryju — APPROVED** [review]
LGTM, but maybe @dbmeneses should also have a look?

**Review by dbmeneses — APPROVED** [review]
LGTM

**Comment by serhat-yenican-sonarsource** [comment]
Thanks a lot for the quick review! @henryju and @dbmeneses

**Pull request merged**